public/app/features/dashboard/components/Inspector/PanelInspectActions.tsx

```diff
@@ -0,0 +1,82 @@
+import { PanelModel } from '../../state';
+
+export interface PanelInspectActionProps {
+  panel: PanelModel;
+  data?: PanelData;
+}
+
+export interface PanelInspectAction {
+  title: string;
+  description: string;
+
+  component: ComponentType<PanelInspectActionProps>;
+}
+
+export interface PanelInspectActionSupplier {
+  getActions: (panel: PanelModel) => PanelInspectAction[] | null;
+}
+
+const dummySupplier: PanelInspectActionSupplier = {
+  getActions: (panel: PanelModel) => {
+    return [
+      {
+        title: 'Do something',
+        description: 'that thingy',
+        // eslint-disable-next-line react/display-name
+        component: ({ panel }) => {
+          return (
+            <div>
+              DO THING ONE. Panel: <pre>{JSON.stringify(panel.targets)}</pre>
+            </div>
+          );
+        },
+      },
+      {
+        title: 'Do something else',
+        description: 'another thing',
+        // eslint-disable-next-line react/display-name
+        component: ({ panel }) => {
+          return (
+            <div>
+              DO THING TWO. Panel: <pre>{JSON.stringify(panel.targets)}</pre>
+            </div>
+          );
+        },
+      },
+    ];
+  },
+};
+
+(window as any).grafanaPanelInspectActionSupplier = dummySupplier;
+
+import React, { ComponentType } from 'react';
+import { PanelData } from '@grafana/data';
+
+interface InspectActionsTabProps {
+  panel: PanelModel;
+  data?: PanelData;
+}
+
+export const InspectActionsTab: React.FC<InspectActionsTabProps> = ({ panel, data }) => {
+  const supplier = (window as any).grafanaPanelInspectActionSupplier as PanelInspectActionSupplier;
```

Comment:
not sure how this is intended to be used, is it just a single plugin that can set this function and add actions? I would have assumed we made this into singleton registry in grafana/data that plugins can add actions to, like a inspectActionsRegistry 

But maybe we want this to be a quick and hidden extension specifically for the ML app only?

Reply:
this is a quick thing to get going with ML plugin, and we'll come up with a more elegant solution afterwards

Reply:
> But maybe we want this to be a quick and hidden extension specifically for the ML app only?

Yes -- I do not think we are prepared to push an API into `@grafana/runtime|data` -- we just need something to get the ML app working quickly.  When we publish an API we should consider how/if it can interact with the panel menu also -- but that was harder than I am able to support for 8.0 :)